### PR TITLE
chore: format with prettier and add prettier stuff

### PR DESCRIPTION
**Pull request by EdwinChang24**

### CODE REVIEW

File: .github/workflows/prettier-check.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Check formatting with Prettier
+
+on: push
```
Comment (Sophon96) [inline .github/workflows/prettier-check.yml]: Can we have this run on pull requests? It doesn't make sense to check it after it's merged.

**Review by Sophon96 — CHANGES_REQUESTED** [review]
Just a small change needed. Looks good otherwise!

**Review by Sophon96 — APPROVED** [review]
LGTM!

File: pnpm-lock.yaml
Comment (Sophon96) [inline pnpm-lock.yaml:1]: Wait bruh add this to `.prettierignore`

**Review by Sophon96 — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml
Comment (Sophon96) [reply pnpm-lock.yaml:1]: And then can you reset the lockfile to `main` and rerun `pnpm install`?

Comment (EdwinChang24) [reply pnpm-lock.yaml:1]: I symlinked `.prettierignore` to `.gitignore` :skull:

File: .prettierignore
Comment (Sophon96) [inline .prettierignore:1]: This is certainly a unique approach

**Review by Sophon96 — APPROVED** [review]

**Pull request merged**